metis-indexing/src/main/java/eu/europeana/indexing/tiers/RecordTierCalculationViewGenerator.java

```diff
@@ -0,0 +1,91 @@
+package eu.europeana.indexing.tiers;
+
+import eu.europeana.corelib.solr.bean.impl.FullBeanImpl;
+import eu.europeana.indexing.exception.IndexingException;
+import eu.europeana.indexing.exception.TierCalculationException;
+import eu.europeana.indexing.fullbean.RdfToFullBeanConverter;
+import eu.europeana.indexing.solr.EdmLabel;
+import eu.europeana.indexing.tiers.view.RecordTierCalculationSummary;
+import eu.europeana.indexing.tiers.view.RecordTierCalculationView;
+import eu.europeana.indexing.utils.RdfTierUtils;
+import eu.europeana.indexing.utils.RdfWrapper;
+import eu.europeana.indexing.utils.SolrTier;
+import eu.europeana.metis.schema.convert.RdfConversionUtils;
+import eu.europeana.metis.schema.convert.SerializationException;
+import eu.europeana.metis.schema.jibx.RDF;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class RecordTierCalculationViewGenerator {
+
+  private final String europeanaId;
+  private final String providerId;
+  private final String stringRdf;
+
+  public RecordTierCalculationViewGenerator(String europeanaId, String providerId, String stringRdf) {
```

Comment:
I think these parameters can be final ```RecordTierCalculationViewGenerator(final String europeanaId, final String providerId, final String stringRdf)``` since they are not being changed in the class.

Reply:
Although in favor of the final keyword, I try not to over use it because I find it pollutes the code and it becomes less readable especially for simple methods where it's clear what they are doing..
Furthermore, we as developers should expect(in almost all cases) that a parameter won't change inside a method, unless explicitly defined that it can change.
In fact I'm curious why java doesn't have all parameters final by default and the developer only define the ones that wants them changeable(so the reversion of what is available now)..

Talking about this specific case I won't add the final since this is a simple constructor.

Reply:
Yep, I remember in a previous experience I had, was sonarqube rule in place in favor / promote inmutability (functional programming) whenever was possible. The advantage of using ```final``` is that it reduces the mental load that the value / reference is never altered later on. So developers need not pay attention to modifications in order to understand the computation. But also understand your point in the sense that it will pollute the code. It something to keep in balance.